### DISCUSSION TITLE
Wear OS complications accesibility improvement

### DISCUSSION THREAD

**Discussion by Asmilex** [NightscoutFoundation/xDrip#2820] [Ideas]
Hello! I've recently bought a Galaxy Watch 5, so I've been playing quite a bit with the xDrip companion. 

I have a suggestion for the complication. It seems that newer Wear OS versions (or maybe Samsungs' One UI?) expect the complication title to be the main focus, leaving the text for small details. The current implementation of xDrip does the following:

```java
// CustomComplicationProviderService.java
// ...
case ComplicationData.TYPE_SHORT_TEXT:

    final ComplicationData.Builder builder = new ComplicationData.Builder(ComplicationData.TYPE_SHORT_TEXT)
            .setShortText(ComplicationText.plainText(numberText))
            .setTapAction(complicationPendingIntent);

    UserError.Log.d(TAG, "TYPE_SHORT_TEXT Current complication state:" + state);
    switch (state) {
        case DELTA:
            builder.setShortTitle(ComplicationText.plainText(getDeltaText(bgReading, is_stale)));
            break;
        case AGO:
            builder.setShortTitle(ComplicationText.plainText(niceTimeSinceBgReading(bgReading)));
            break;
        default:
            builder.setShortTitle(ComplicationText.plainText("ERR!"));

```

which results in this:

![photo_2023-05-07_20-35-10](https://user-images.githubusercontent.com/32983998/236696461-0dda9bab-bf54-4329-9167-cc0b1e306cee.jpg)

One simple way of fixing this would be swapping them. I don't know how many uses cases would affect this, since I've seen other Wear OS systems do the opposite. 

I've thought about creating a new complication provider focusing only on this behavior. Notably, we could move the trend arrow to the Icon as a [MonochromaticIcon](https://developer.android.com/reference/androidx/wear/watchface/complications/data/ShortTextComplicationData.Builder#setMonochromaticImage(androidx.wear.watchface.complications.data.MonochromaticImage)). This would leave space for bigger text. 

What do you think?

**Comment by gdc27health** [comment]
The complication as it is passed to faces built with Watch Face Studio is fine the way it is.  In WFS the developer can adjust the size of both the text and title fields.

What would be really really nice is to have a complication or ability to use the data in a graph.  The Watchmaker app has graphs, but are paid.

**Reply by gdc27health** [reply]
![Screenshot_20230522_102348_sysui (1)](https://github.com/NightscoutFoundation/xDrip/assets/129467702/2ed70ed2-bb00-4412-98e9-259ae12ed104)

**Reply by gdc27health** [reply]
If the user taps on the complication the smaller field changes between the time and the amount the glucose levels changed
